editor/effects.js

```diff
@@ -27,17 +28,30 @@ export default {
 			...edits,
 			content: serialize( getBlocks( state ) ),
 		};
+		const transactionId = uniqueId();
 
 		if ( ! isNew ) {
 			toSend.id = postId;
 		}
 
-		dispatch( { type: 'CLEAR_POST_EDITS' } );
+		dispatch( {
+			type: 'CLEAR_POST_EDITS',
+			optimist: { type: BEGIN, id: transactionId },
+		} );
+		dispatch( {
+			type: 'UPDATE_POST',
+			edits: toSend,
+			optimist: { id: transactionId },
+		} );
 		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
 			dispatch( {
 				type: 'REQUEST_POST_UPDATE_SUCCESS',
-				post: newPost,
 				isNew,
+				optimist: { type: COMMIT, id: transactionId },
+			} );
+			dispatch( {
+				type: 'UPDATE_POST',
+				edits: newPost,
```

Comment:
This doesn't seem like something we should _merge_ into our representation of a post, per the behavior of the reducer, but instead reset / replace with. Maybe a separate `RESET_POST` that has the same behavior as the `currentPost` handler for `RESET_BLOCKS`? In fact, I'm thinking maybe we should separate out a `RESET_POST` from `RESET_BLOCKS` anyways, because the `post` property here feels kinda awkwardly included anyways:

https://github.com/WordPress/gutenberg/blob/6dea0df/editor/index.js#L49-L53

Maybe instead:

```js
store.dispatch( { type: 'RESET_POST', post } );
store.dispatch( { type: 'RESET_BLOCKS', blocks: wp.blocks.parse( post.content.raw ) } );
```

Or maybe we don't need `RESET_BLOCKS` at all, and consolidate further to a single `RESET_POST` which the block reducers listen for and `parse` upon.

(Forgive my spew of brain-thoughts 😄 )